addons/docs/src/frameworks/angular/compodoc.ts

```diff
@@ -97,11 +97,20 @@ const extractTypeFromValue = (defaultValue: any) => {
   return defaultValue || valueType === 'boolean' ? valueType : null;
 };
 
-const extractEnumValues = (compodocType: any) => {
+const extractEnumValues = (compodocType: any): string[] => {
   if (typeof compodocType !== 'string' || compodocType.indexOf('|') === -1) {
-    return null;
+    return [];
   }
-  return compodocType.split('|').map((value) => JSON.parse(value));
+  return compodocType
+    .split('|')
+    .map((value) => {
+      try {
+        return JSON.parse(value);
+      } catch (e) {
+        return null;
```

Comment:
@matheo what kinds of values are causing the exception here? can you share some examples? filtering them out like this seems like sweeping things under the rug, and maybe we can do better?

Reply:
compodocType equal to `string | number | Date`, or `string | string[]` or `string | null`
those unions shouldn't be processed as enums tho, and yes, probably it can be better figuring out the `string` type for this kind of stuff tho, but I guess it's ok to setup the controls manually for those objects.

Reply:
OK, makes sense. How about this proposal: if any of the JSON.parse fails, we return `null` and that makes the control an `object` type, which is super flexible. If the user wants to customize it, they can.

Reply:
done!